src/core/python/xml_v.cpp

```diff
@@ -362,6 +364,12 @@ void parse_dictionary(DictParseContext &ctx,
             continue;
         } catch (const pybind11::cast_error &) { }
 
+        // Try to cast to TensorXf
+        try {
+            props.set_tensor_handle(key, std::make_shared<TensorXf>(value.template cast<TensorXf>()));
```

Comment:
Another option would be to try to convert/cast it to the current variant.

Reply:
This _just works_:tm:

In `properites_v`: 
The [overload resolution order](https://pybind11.readthedocs.io/en/stable/advanced/functions.html?highlight=resolution%20order#overload-resolution-order) will attempt a conversion to `TensorXf` of the current `Properties` variant. No matter what the argument is, this is all handled by `tensor_init()` (even changes of variants).

In `xml_v.cpp`:
The caster will attempt to use the `TensorXf` constructor if the current type & variant does not match, in which case it will also resort to `tensor_init()`.

One problem that might exist is similar to https://github.com/mitsuba-renderer/mitsuba3/issues/676 I believe, where the `Properties` type itself doesn't update its bindings.

Admittedly, this is mostly pybind11 magic and took me a while to figure out why it works...

Reply:
Awesome! Thanks for the explanations 😉